### PR TITLE
Refactoring work-in-progress

### PR DESCRIPTION
**Pull request by nyalldawson**
(do not merge)

### CODE REVIEW

**Comment by nyalldawson** [comment]
@ghtmtt this part should be ready to merge now -- did you want to review?

**Comment by ghtmtt** [comment]
having a look right now. Will comment if needed or directly merge it. Thanks!

File: DataPlotly/core/plot_factory.py [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+from DataPlotly.core.plot_types import *  # pylint: disable=W0401,W0614
+
+
+class PlotFactory:  # pylint:disable=too-many-instance-attributes
+    """
+    Plot factory which creates Plotly Plot objects
+
+    Console usage:
+    # create the object
+    p = Plot(plot_type, plot_properties, layout_properties)
```
[the remark this answers is not in the document]
Comment (nyalldawson) [reply DataPlotly/core/plot_factory.py]: Done

File: DataPlotly/core/plot_types/plot_type.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+"""
+
+from plotly import graph_objs
+from qgis.PyQt.QtCore import QCoreApplication
+
+
+class PlotType:
```
[the remark this answers is not in the document]
Comment (nyalldawson) [reply DataPlotly/core/plot_types/plot_type.py]: ...left as an exercise for the reader ;)

File: DataPlotly/core/plot_types/bar_plot.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+Bar plot
+
+.. note:: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+"""
+
+import os
+from plotly import graph_objs
```
[the remark this answers is not in the document]
Comment (nyalldawson) [reply DataPlotly/core/plot_types/bar_plot.py]: I tried that - but it flags the plot types (e.g. `Bar`) as an error (could not find definition) in the linting. There's something tricky going on in the graph_objects module which is throwing out the linter, so if it's ok with you can we leave it this way?

[the remark this answers is not in the document]
Comment (ghtmtt) [reply DataPlotly/core/plot_types/bar_plot.py]: +1 no objections

**Pull request merged**